src/server/server.js

```diff
@@ -124,13 +122,29 @@ app.post('/callback', (req, res) => {
         });
 });
 
+// Protected routes
+app.use('/*', (req, res, next) => {
+    if (
+        process.env.NODE_ENV === 'development' ||
+        authsupport.stillValid(req.session.spadeToken)
+    ) {
+        next();
+    } else {
+        console.log(
+            `no valid session found for ${req.headers['x-forwarded-for'] ||
+                req.connection.remoteAddress}`
+        );
+        res.redirect('/login');
+    }
+});
+
+app.use('/static', express.static('dist'));
+
 app.get('/whoami', (req, res) => {
     if (process.env.NODE_ENV === 'development') {
         res.send({ name: `Sara Saksbehandler` });
-    } else if (req.cookies['speil']) {
-        res.send({ name: `${tokendecoder.username(req.cookies['speil'])}` });
     } else {
-        res.sendStatus(401);
+        res.send({ name: `${authsupport.username(req.cookies['speil'])}` });
```

Comment:
Klarer ikke helt å finne angrepsscenariet, men føler litt sånn "hmmm.." rundt echo-ing av ikke-validerte data (?) :-/

Reply:
Dette endepunktet ser ikke ut til å ha livets rett lenger, da klienten vel selv kan fiske ut brukernavnet heller enn å gjøre et kall for å få dekodet infoen?

Reply:
Men den får ikke fisket ut fra cookien hvis cookien er httpOnly(?) da

Reply:
Den er http only

Reply:
Det er den jo, ja. Vi har jo et sesjonsobjekt også? Men vi ønsker kanskje ikke å ta vare på ymse data der.